### PR TITLE
Merge custom avatar

### PR DESCRIPTION
**Pull request by qjgn**
### 50c7573 Applied [CustomAvatar](https://github.com/qualisys/Qualisys-Unity-SDK/pull/18) branch to the latest master
The custom avatar feature provides a way of controlling non humanoid rigged models in Unity.
Previously we only supported retargeting humanoid rigs with the RTSkeleton script. 
The new RTGenericSkeleton lets you control any rigged models with the QTMs skeleton stream.
Using the RTGenericSkeleton looks the best when you have a rigged model that fits the source skeleton perfectly. This is because it does not do any retargeting magic.

Changes from the original PR
- Put the skeleton segment calculations back into the RTStreamThread. Instead of having two separate ways of transforming the segments, we are back to one. So no need for leaking the transformations methods onto the RT* behaviours. This also lets us do all the QTM->Unity Coordinate conversions before the data leaves the worker thread.
- Added hierarchy debug script for visualizing segment rotations
- Separated the behavior into two script RTSkeleton, RTGenericSkeleton 

### 32ac087 Refactored Qtm RHS to Unity LHS conversions
- Added extensions to the RTClientSDK position and rotation types for converting to Unity Left hand system coordinates.
- Refactored the old conversions. 
- Removed unused rotation helper code.

### efc3426 Remember last used QTM server
The RTGUI tab now remembers the last used server, even if the list changes!

### Todo:

- ~Add the rest of the component rotations from CustomAvatar~
- ~Figure out why the humanoid mapping root bone needs to be rotated 180 degrees. Is this adjustment universal? If not is it possible to calculate on the fly?~
- (skip for now) ~Investigate alternative Generic mecanim mapping using the unity API https://docs.unity3d.com/ScriptReference/AvatarBuilder.html~
- ~Add demo files~

### Test with QAvatar
generic_and_humanoid_uniform_input
### Test with Cricket Player and Bat (50% speed)
cricket_player

### CODE REVIEW

File: Qualisys/QTM-Unity-Realtime-Streaming/Helpers/DebugHierarchyRotations.cs [hunk 1]
```diff
@@ -0,0 +1,34 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class DebugHierarchyRotations : MonoBehaviour
```
Comment (helmesjo) [inline Qualisys/QTM-Unity-Realtime-Streaming/Helpers/DebugHierarchyRotations.cs]: Should this be in the same namespace as the rest?

File: Qualisys/QTM-Unity-Realtime-Streaming/Streaming/RTSkeleton.cs [hunk 2]
```diff
@@ -107,9 +101,9 @@ private void BuildMecanimAvatarFromQtmTPose()
...
                 name = this.SkeletonName,
                 position = Vector3.zero,
-                rotation = Quaternion.identity,
+                rotation = Quaternion.AngleAxis(180, Vector3.up), //TODO: Is it possible to calculate this?
                 scale = Vector3.one,
-            });
+            }); 
 
             // Create remaining T-Pose bone definitions from Qtm segments
             foreach (var segment in mQtmSkeletonCache.Segments.ToList())
```
Comment (helmesjo) [inline Qualisys/QTM-Unity-Realtime-Streaming/Streaming/RTSkeleton.cs]: Is it really necessary to first convert to a list here?

File: Qualisys/QTM-Unity-Realtime-Streaming/Streaming/RTSkeleton.cs [hunk 3]
```diff
@@ -107,9 +101,9 @@ private void BuildMecanimAvatarFromQtmTPose()
             {
                 name = this.SkeletonName,
                 position = Vector3.zero,
-                rotation = Quaternion.identity,
+                rotation = Quaternion.AngleAxis(180, Vector3.up), //TODO: Is it possible to calculate this?
```
Comment (helmesjo) [inline Qualisys/QTM-Unity-Realtime-Streaming/Streaming/RTSkeleton.cs]: "`TODO`"

File: Qualisys/QTM-Unity-Realtime-Streaming/Streaming/RTSkeleton.cs (hunk 2 above)
Comment (qjgn) [reply Qualisys/QTM-Unity-Realtime-Streaming/Streaming/RTSkeleton.cs]: It's not

File: Qualisys/QTM-Unity-Realtime-Streaming/Helpers/DebugHierarchyRotations.cs (hunk 1 above)
Comment (qjgn) [reply Qualisys/QTM-Unity-Realtime-Streaming/Helpers/DebugHierarchyRotations.cs]: Yes

File: Qualisys/QTM-Unity-Realtime-Streaming/Streaming/RTSkeleton.cs (hunk 3 above)
Comment (qjgn) [reply Qualisys/QTM-Unity-Realtime-Streaming/Streaming/RTSkeleton.cs]: In QTM default poses are facing +Y which becomes -Z in Unity.
We rotate 180 degrees to match unity forward.

**Review by EmilRosenquist — APPROVED** [review]

**Review by helmesjo — APPROVED** [review]

**Review by Thingvold — APPROVED** [review]
Yes!

**Pull request merged**